src/libopenrave/kinbodystatesaver.cpp

```diff
@@ -93,6 +93,12 @@ void KinBody::KinBodyStateSaver::_RestoreKinBody(boost::shared_ptr<KinBody> pbod
         for (const GrabbedPtr& pGrabbed : _vGrabbedBodies) {
             KinBodyPtr pGrabbedBody = pGrabbed->_pGrabbedBody.lock();
             if( !!pGrabbedBody ) {
+                KinBody::LinkPtr pGrabbingLink(pGrabbed->_pGrabbingLink);
+                if( !!pGrabbingLink && !pbody->GetLink(pGrabbingLink->GetName()) ) {
```

Comment:
Shouldn't we throw exception if either ``pGrabbingLink`` or ``pbody->GetLink(pGrabbingLink->GetName())`` is nullptr?

Reply:
i would do `!pGrabbingLink || !pbody->GetLink(pGrabbingLink->GetName()`

Reply:
@kanbouchou @rdiankov thank you, I updated the code to check `!pGrabbingLink` and remove old exception checks for warnings.